alg_convex_hull.go

```diff
@@ -161,12 +162,17 @@ func sortByPolarAngle(ps []XY) {
 
 	ps = ps[1:] // only sort the remaining points
 	sort.Slice(ps, func(i, j int) bool {
+		// If any point is equal to the anchor point, then always put it first.
+		// This allows those duplicated points to be removed when the results
+		// stack is initiated.
```

Comment:
So this one aims to remove duplicated point, but I think the duplicated point will be removed when doing graham scan, won't it?

Reply:
It's true; duplicated points will be removed by the graham scan. However, there's one edge case -- if the lowest-leftmost point is duplicated, then the naive graham scan wouldn't remove that duplication. That's because the first step of the graham scan is to grab the first two points and push them onto the result stack (but those two points might be duplicates!). By sorting all of the points coincident to the anchor point first, we can easily skip past any duplicate points when initializing the result stack.